lib/Optimizer/Lower/Lower.cpp

```diff
@@ -159,6 +159,76 @@ static void lowerRegressionGradNode(Function *F, CompilationContext &cctx,
                        expG);
 }
 
+static void lowerGemmNode(Function *F, CompilationContext &cctx,
+                          const GemmNode &GN) {
+
+  LOG_SCOPE(F->getLogContext(), "lowerGemm")
+
+  NodeValue A = GN.getA();
+  NodeValue B = GN.getB();
+  NodeValue C = GN.getC();
+  NodeValue Y = GN.getResult();
+  float alpha = GN.getAlpha();
+  float beta = GN.getBeta();
+
+  // Transpose A (if required).
+  if (GN.getTransposeA()) {
+    A = F->createTranspose(DECORATE_NODE_NAME(GN, "TransposeA"), A, {1, 0});
+  }
+
+  // Transpose B (if required).
+  if (GN.getTransposeB()) {
+    B = F->createTranspose(DECORATE_NODE_NAME(GN, "TransposeB"), B, {1, 0});
+  }
+
+  // If Gemm is same as FullyConnected then lower to a FullyConnected node.
+  if (isGemmSameAsFullyConnected(&GN)) {
+    NodeValue newY =
+        F->createFullyConnected(GN.getName().str(), A, B, C, Y.getType());
+    replaceAllUsesOfWith(cctx.loweredInfoMap, GN.getResult(), newY);
+    return;
+  }
+
+  // Create MatMul for A * B.
+  NodeValue newY =
+      F->createMatMul(DECORATE_NODE_NAME(GN, "MatMul"), Y.getType(), A, B);
+
+  // Multiply with alpha (if required).
+  if (alpha != 1.0) {
+    auto *alphaSplat = F->createSplat(DECORATE_NODE_NAME(GN, "AlphaSplat"),
+                                      Y.getType(), alpha);
+    newY = F->createMul(DECORATE_NODE_NAME(GN, "AlphaMul"), alphaSplat, newY);
+  }
+
+  // Check if C operand is used.
+  bool isUsedC = C.getNode();
+  if (isUsedC) {
+    auto *splatC = llvm::dyn_cast<SplatNode>(C.getNode());
+    if (splatC && (splatC->getValue() == 0)) {
+      isUsedC = false;
+    }
+  }
+
+  // Add C (if used).
+  if (isUsedC) {
```

Comment:
nit: Not sure if it'd ever be relevant, but could also skip this if `beta == 0.f`

Reply:
Good point!. I modified with this:
```
bool isUsedC = C.getNode() && (beta != 0.f);
```
Done!